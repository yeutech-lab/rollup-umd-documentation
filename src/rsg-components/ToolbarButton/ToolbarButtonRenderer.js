import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import cx from 'classnames';

export const styles = ({ space, color }) => ({
	button: {
		padding: 2, // Increase clickable area a bit
		color: color.light,
		background: 'transparent',
		transition: 'color 750ms ease-out',
		cursor: 'pointer',
		'&:hover, &:focus': {
			isolate: false,
			color: color.linkHover,
			transition: 'color 150ms ease-in',
		},
		'&:focus': {
			isolate: false,
			outline: [[1, 'dotted', color.linkHover]],
		},
		'& + &': {
			isolate: false,
			marginLeft: space[1],
		},
		// Style react-icons icon passed as children
		'& svg': {
			width: space[3],
			height: space[3],
			color: 'currentColor',
			cursor: 'inherit',
		},
	},
	isSmall: {
		'& svg': {
			width: 14,
			height: 14,
		},
	},
});

export function ToolbarButtonRenderer({
	classes,
	className,
	onClick,
	href,
	title,
	small,
	children,
}) {
	const classNames = cx(classes.button, className, {
		[classes.isSmall]: small,
	});

	if (href !== undefined) {
		return (
			<a href={href} title={title} className={classNames} aria-label={title}>
				{children}
			</a>
		);
	}

	return (
		<button type="button" onClick={onClick} title={title} className={classNames} aria-label={title}>
			{children}
		</button>
	);
}

ToolbarButtonRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	className: PropTypes.string,
	href: PropTypes.string,
	onClick: PropTypes.func,
	title: PropTypes.string,
	small: PropTypes.bool,
	children: PropTypes.node,
};

export default Styled(styles)(ToolbarButtonRenderer);
