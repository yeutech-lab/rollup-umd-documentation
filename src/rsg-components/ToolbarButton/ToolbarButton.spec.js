import React from 'react';
import { ToolbarButtonRenderer, styles } from './ToolbarButtonRenderer';

const props = {
	classes: classes(styles),
	title: 'Pizza button',
};

it('should render a button', () => {
	const actual = shallow(
		<ToolbarButtonRenderer {...props} onClick={() => {}}>
			pizza
		</ToolbarButtonRenderer>
	);

	expect(actual).toMatchSnapshot();
});

it('should render a link', () => {
	const actual = shallow(
		<ToolbarButtonRenderer {...props} href="/foo">
			pizza
		</ToolbarButtonRenderer>
	);

	expect(actual).toMatchSnapshot();
});

it('should pass a class name to a button', () => {
	const actual = shallow(
		<ToolbarButtonRenderer {...props} onClick={() => {}} className="foo-class">
			pizza
		</ToolbarButtonRenderer>
	);

	expect(actual.prop('className')).toBe('button foo-class');
});

it('should pass a class name to a link', () => {
	const actual = shallow(
		<ToolbarButtonRenderer {...props} href="/foo" className="foo-class">
			pizza
		</ToolbarButtonRenderer>
	);

	expect(actual.prop('className')).toBe('button foo-class');
});

it('should render a button with small styles', () => {
	const actual = shallow(
		<ToolbarButtonRenderer {...props} onClick={() => {}} small>
			butterbrot
		</ToolbarButtonRenderer>
	);

	expect(actual.prop('className')).toBe('button isSmall');
});
