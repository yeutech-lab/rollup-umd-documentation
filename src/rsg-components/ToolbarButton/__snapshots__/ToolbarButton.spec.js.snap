// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a button 1`] = `
<button
  aria-label="Pizza button"
  className="button"
  onClick={[Function]}
  title="Pizza button"
  type="button"
>
  pizza
</button>
`;

exports[`should render a link 1`] = `
<a
  aria-label="Pizza button"
  className="button"
  href="/foo"
  title="Pizza button"
>
  pizza
</a>
`;
